sdks/cpp/common/include/ParamAccessor.h

```diff
@@ -462,8 +456,7 @@ class ParamAccessor {
                 }
                 variantInfo.members.at(variant).wrapSetter(sp.get(), &src);
             } else {
-                typename std::remove_const<typename std::remove_reference<decltype(src)>::type>::type x;
```

Comment:
I added const to the parameters of getKindCase() so that a const argument can be passed directly without the need of remove_const. It also still works when passed non-const arguments.

Reply:
what about when passed const ref arguments?

Reply:
getKindCase was already pass by reference. I have tested it and all test still pass.